.github/workflows/test.yaml

```diff
@@ -87,7 +94,23 @@ jobs:
         with:
           tool: cargo-make
 
+      # Code coverage is generated only for amd64 because cargo-llvm-cov
+      # does not currently support Windows on arm64.
+      # See issue: https://github.com/taiki-e/cargo-llvm-cov/issues/436
+      - name: Run Cargo Test with Coverage (workspace)
+        if: matrix.runner.arch != 'arm64'
+        run: cargo +${{ matrix.rust_toolchain }} llvm-cov --workspace --codecov --output-path target/codecov.info --locked --profile ${{ matrix.cargo_profile }} --all-features
+
+      - name: Upload Coverage to Codecov
+        if: matrix.runner.arch != 'arm64'
+        uses: codecov/codecov-action@v5
+        with:
+          files: target/codecov.info
+          token: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
What would be the behaviour of this action be in case this token is not available? Would it skip the upload with a warning (or) fail the workflow?

IMO, Failing the workflow would be odd for the forked repos, as this step is not critical for them. It would be better to treat it as a warning if that's not currently the behaviour.

Reply:
That's a good point. I have never relied on workflows running in my own fork. If I need to get some tests run I open a draft PR on the upstream repo instead. That said, others may have different preferences so let me try to address this.

Reply:
How about skipping upload when secrets.CODECOV_TOKEN token is not available, instead of doing it based on the repository? This way, it may still work on forked repos if someone has set their personal token.